include/supervision.h

```diff
@@ -105,9 +105,30 @@ extern unsigned char sv_nmi_counter;
 extern unsigned char sv_timer_irq_counter;
 extern unsigned char sv_timer_dma_counter;
 
+/* Masks for joy_read */
+#define JOY_UP_MASK     0x08
+#define JOY_DOWN_MASK   0x04
+#define JOY_LEFT_MASK   0x02
+#define JOY_RIGHT_MASK  0x01
+#define JOY_BTN_1_MASK  0x20
```

Comment:
That JOY_BTN_x_MASK thingy is by no means obligatory. As far as I can see from Supervision pics it's likely no appropriate. I personally would rather define A, B, START and SELECT right away based on the numeric values.

Reply:
What do you mean? I don't see any difference to the NES...

Reply:
Yes, you are right, please just ignore my comment ;-)